File: packages/app/src/services/software/definitions/variants/discordPtb.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'discord-ptb';
	public name = 'Discord PTB';
	public category = [SoftwareCategory.Social];
	public downloadName = 'DiscordPTBSetup.exe';
	public icon = 'discord.png';
	public homepage = 'https://ptb.discord.com';
	public parent = 'discord';

	public async resolveDownloadUrl() {
		return ok('https://ptb.discord.com/api/downloads/distributions/app/installers/latest?channel=ptb&platform=win&arch=x64');
	}
}

File: src/lib/software/variants/discord_ptb.py
from src.lib.software import BaseSoftware

class DiscordPtb(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'DiscordPTBSetup.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://ptb.discord.com/api/downloads/distributions/app/installers/latest?channel=ptb&platform=win&arch=x64')
